src/appdistribution/client.ts

```diff
@@ -69,32 +83,36 @@ export class AppDistributionClient {
   }
 
   async pollReleaseIdByHash(hash: string, retryCount = 0): Promise<string> {
-    try {
-      return await this.getReleaseIdByHash(hash);
-    } catch (err) {
+    const uploadStatus = await this.getUploadStatus(hash);
+    if (uploadStatus.status === UploadStatus.IN_PROGRESS) {
       if (retryCount >= AppDistributionClient.MAX_POLLING_RETRIES) {
-        throw new FirebaseError(`failed to find the uploaded release: ${err.message}`, { exit: 1 });
+        throw new FirebaseError(
+          "failed to fetch release information: polling timeout exceeded, please try again",
```

Comment:
Let’s have this more generic (to not show our cards that we're actually polling), similar to the gradle plugin: `it took longer than expected to process your binary, please try again`